### PR TITLE
Repl input changes and less signficant file size

### PR DESCRIPTION
**Pull request by SCLorentz**
I've added some functionalities to the repl, like, command line history and confirmations to exit (ctrl+c).

I've changed some settings as well, and now the final compiled file uses less space than before (28,1 mb --> 2,3 mb)

### CODE REVIEW

**Comment by SCLorentz** [comment]
wat: https://www.destroyallsoftware.com/talks/wat

File: Cargo.toml [hunk 1]
```diff
@@ -8,3 +8,10 @@ edition = "2021"
 [dependencies]
 clap = { version = "4.3.8", features = ["derive"] }
 lazy-regex = "2.5.0"
+rustyline = "=5.0.2"
```
Comment (PokeJofeJr4th) [inline Cargo.toml:11]: Why is this a specific version?

File: Cargo.toml [hunk 2]
```diff
@@ -8,3 +8,10 @@ edition = "2021"
...
+dialoguer = "0.11.0"
+
+[profile.dev]
+opt-level = "z"
```
Comment (PokeJofeJr4th) [inline Cargo.toml:15]: Why are you choosing to optimize for binary size rather than say performance or comp times?

File: src/main.rs [hunk 3]
```diff
@@ -1,11 +1,22 @@
 #![warn(clippy::pedantic, clippy::nursery)]
 
+use std::alloc::System;
+
+#[global_allocator]
+static A: System = System;
```
Comment (PokeJofeJr4th) [inline src/main.rs:6]: Why specify the global allocator? As far as I can tell this is implicit

**Review by PokeJofeJr4th — COMMENTED** [review]
Overall I really like it, just a couple choices I'd like to know more about before merging

File: Cargo.toml (hunk 1 above)
Comment (SCLorentz) [reply Cargo.toml:11]: the command line provided to install this library in the crates website was 'rustyline = "=5.0.2"', I just use it exactly how it was provided. Any way, I found another library that I think that may be better, but idk, I will look more about it.

File: Cargo.toml (hunk 2 above)
Comment (SCLorentz) [reply Cargo.toml:15]: This software is already fast, and the file size was big. So I think is better to sacrifice some optmization in exchange for a small file. And, about the compilation time, well this is a process that doesn't need to be done all the time, and some people can even compile it just once, so, again, is for the best experience in my opinion.

File: src/main.rs (hunk 3 above)
Comment (SCLorentz) [reply src/main.rs:6]: I was following some recommendations that i've found in the https://github.com/johnthagen/min-sized-rust repository. It did work on a project that i was doing, so I use it again. If this is unnecessary, forgive me, I'm really young, and I'm just trying to learn more of rust, so I will search more about rust optmizations methods.

**Comment by PokeJofeJr4th** [comment]
I got all of my little changes in; this is now ready to merge unless we decide to switch away from rustyline

**Review by SCLorentz — COMMENTED** [review]
Thank you, I appreciate your suggestions! :)

**Pull request merged**